src/main/java/hudson/plugins/clover/CloverPublisher.java

```diff
@@ -224,6 +280,54 @@ public String invoke(File file, VirtualChannel virtualChannel) throws IOExceptio
         }
     }
 
+    private void processCloverXml(Run<?, ?> build, FilePath workspace, TaskListener listener, FilePath
+            coverageReport, FilePath buildTarget) throws InterruptedException {
+
+        String workspacePath = "";
+        try {
+            workspacePath = workspace.act(new SlaveToMasterFileCallable<String>() {
+                public String invoke(File file, VirtualChannel virtualChannel) throws IOException {
+                    try {
+                        return file.getCanonicalPath();
+                    } catch (IOException e) {
+                        return file.getAbsolutePath();
+                    }
+                }
+            });
+        } catch (IOException e) {
+        }
+        if (!workspacePath.endsWith(File.separator)) {
+            workspacePath += File.separator;
+        }
+
+        final File cloverXmlReport = getCloverXmlReport(build);
+        if (cloverXmlReport.exists()) {
+            listener.getLogger().println("Publishing Clover coverage results...");
+            ProjectCoverage result = null;
+            try {
+                result = CloverCoverageParser.parse(cloverXmlReport, workspacePath);
+            } catch (IOException e) {
+                Util.displayIOException(e, listener);
+                e.printStackTrace(listener.fatalError("Unable to copy coverage from " + coverageReport + " to " + buildTarget));
+                build.setResult(Result.FAILURE);
```

Comment:
Yes sure, I only mention this because in my case the diff appears as new (and also for me all the code is new). I was only wondering this as an improvement but nothing blocking. But at the same time I would say that if we adopt this policy, then I think that for example JaCoCo or JUnit plugins should follow the same approach.